### PR TITLE
fmt: update to 8.0.1. [ci skip]

### PR DESCRIPTION
**Pull request by skmpz**
#### General
- [ ] This is a new package and it conforms to the [quality requirements](https://github.com/void-linux/void-packages/blob/master/Manual.md#quality-requirements)

#### Have the results of the proposed changes been tested?
- [x] I use the packages affected by the proposed changes on a regular basis and confirm this PR works for me
- [ ] I generally don't use the affected packages but briefly tested this PR

#### Does it build and run successfully? 
(Please choose at least one native build and, if supported, at least one cross build. More are better.)
- [x] I built this PR locally for my native architecture, (x86_64)
- [x] I built this PR locally for these architectures (if supported. mark crossbuilds):
  - [x] i686
  - [x] x86_64-musl
  - [x] armv7l (cross x86_64)
  - [x] aarch64-musl (cross x86_64-musl)
  - [x] armv6l-musl (cross x86_64-musl)

Packages built (compile+tests):
- [x] fmt
- [x] spdlog
- [x] cherrytree
- [x] cryfs
- [x] gerbera
- [x] Waybar
- [x] mkvtoolnix
- [x] nheko
- [x] 0ad
- [x] kodi

### CODE REVIEW

**Comment by ericonr** [comment]
Nice! Let me know if you hit any bumps.

**Comment by skmpz** [comment]
@ericonr everything builds now. For kodi i added just the minimal needed to build with fmt8. Please have a look :)

**Comment by ericonr** [comment]
Any reason you skipped CI? I figure cross might be able to complete, at least.

**Comment by ericonr** [comment]
I will try building them locally and see how everything goes.

**Comment by ericonr** [comment]
Ok, this is missing kodi-rpi.

**Comment by ericonr** [comment]
I would reword the 0ad commit message too. Just adding the patch wouldn't require a rebuild, it's happening because of fmt.

**Comment by ericonr** [comment]
Tried adding your kodi patch to kodi-rpi, but it wasn't enough:

```
../xbmc/utils/StringUtils.h:105:32: error: no matching function for call to 'format(const wstring&, remove_rvalue_reference<const wchar_t*&&>::type)'
  105 |     auto result = ::fmt::format(fmt, EnumToInt(std::forward<Args>(args))...);
      |                   ~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

We should maybe consider removing kodi-rpi or making it a metapackage that pulls in `kodi`. Otherwise it's going to be piling up on fixes and more fixes.

**Comment by ericonr** [comment]
Pushed rebased and reworded commits into the PR. Things seem to be building ok and working fine; what we need to fix now is just the kodi-rpi thing.

**Comment by ericonr** [comment]
Thanks a lot!
